### PR TITLE
Code optimized by rsu

### PR DESCRIPTION
**Pull request by rsubrama83**

### CODE REVIEW

**Review by InDieTasten — CHANGES_REQUESTED** [review]
Waiting for previous code changes to be tested before merging this

**Review by fde-lmag — COMMENTED** [review]
PlatformCallController.OnNotificationRequestAsync
=> AuthenticationProvider.ValidateInboundRequestAsync
=> JwtSecurityTokenHandler.ValidateToken
=> throws Exception
IDX12709: CanReadToken() returned false. JWT is not well formed.
The token needs to be in JWS or JWE Compact Serialization Format. (JWS): 'EncodedHeader.EndcodedPayload.EncodedSignature'. (JWE): 'EncodedProtectedHeader.EncodedEncryptedKey.EncodedInitializationVector.EncodedCiphertext.EncodedAuthenticationTag'.

**Review by fde-lmag — COMMENTED** [review]
Seems these changes after merging with main wont work
The branch as is runs into a 401 (JWT parse exception that was fixed on main)
The branch merged with main seems to not work properly (no requests are coming in after accepting a call)

File: Samples/PublicSamples/RecordingBot/src/RecordingBot.Services/Bot/CallHandler.cs [hunk 1]
```diff
@@ -183,7 +181,7 @@ private async void CallOnUpdated(ICall sender, ResourceEventArgs<Call> e)
                 }
             }
 
-            if ((e.OldResource.State == CallState.Established) && (e.NewResource.State == CallState.Terminated))
+            if (e.OldResource.State == CallState.Established && e.NewResource.State == CallState.Terminated)
```
Comment (1fabi0) [inline Samples/PublicSamples/RecordingBot/src/RecordingBot.Services/Bot/CallHandler.cs:184]: To check for newResource state is enough

File: Samples/PublicSamples/RecordingBot/src/RecordingBot.Services/Http/Controllers/PlatformCallController.cs [hunk 2]
```diff
@@ -65,8 +67,9 @@ public PlatformCallController(IGraphLogger logger, IBotService botService)
 
             RequestValidationResult result;
 
-                var httpRequestMessage = new System.Net.Http.HttpRequestMessage();
-                httpRequestMessage.Headers.Authorization = new System.Net.Http.Headers.AuthenticationHeaderValue("Bearer", Request.Headers.Authorization);
+                var httpRequestMessage = new HttpRequestMessage();
+                httpRequestMessage.Headers.Authorization = new AuthenticationHeaderValue("Bearer", Request.Headers.Authorization);
```
Comment (1fabi0) [inline Samples/PublicSamples/RecordingBot/src/RecordingBot.Services/Http/Controllers/PlatformCallController.cs:71]: Request.Headers.Authorization needs to be splitted from Bearer( see how it is done in main Branch)

**Review by 1fabi0 — CHANGES_REQUESTED** [review]

**Pull request closed without merging**